sql/core/src/test/resources/sql-tests/inputs/pgSQL/aggregates_part3.sql

```diff
@@ -0,0 +1,287 @@
+
+-- select * from agg_view1;
+-- select pg_get_viewdef('agg_view1'::regclass);
+
+-- create or replace view agg_view1 as
+--   select aggfns(a,a,c order by b)
+--     from (values (1,3,'foo'),(0,null,null),(2,2,'bar'),(3,1,'baz')) v(a,b,c);
+
+-- select * from agg_view1;
+-- select pg_get_viewdef('agg_view1'::regclass);
+
+-- create or replace view agg_view1 as
+--   select aggfns(a,b,c order by c using ~<~)
+--     from (values (1,3,'foo'),(0,null,null),(2,2,'bar'),(3,1,'baz')) v(a,b,c);
+
+-- select * from agg_view1;
+-- select pg_get_viewdef('agg_view1'::regclass);
+
+-- create or replace view agg_view1 as
+--   select aggfns(distinct a,b,c order by a,c using ~<~,b)
+--     from (values (1,3,'foo'),(0,null,null),(2,2,'bar'),(3,1,'baz')) v(a,b,c),
+--          generate_series(1,2) i;
+
+-- select * from agg_view1;
+-- select pg_get_viewdef('agg_view1'::regclass);
+
+-- drop view agg_view1;
+
+-- incorrect DISTINCT usage errors
+
+-- select aggfns(distinct a,b,c order by i)
+--   from (values (1,1,'foo')) v(a,b,c), generate_series(1,2) i;
+-- select aggfns(distinct a,b,c order by a,b+1)
+--   from (values (1,1,'foo')) v(a,b,c), generate_series(1,2) i;
+-- select aggfns(distinct a,b,c order by a,b,i,c)
+--   from (values (1,1,'foo')) v(a,b,c), generate_series(1,2) i;
+-- select aggfns(distinct a,a,c order by a,b)
+--   from (values (1,1,'foo')) v(a,b,c), generate_series(1,2) i;
+
+-- [SPARK-27978] We use concat_ws(delimiter, collect_list(expression)) to rewrite string_agg
+-- string_agg tests
+select concat_ws(',', collect_list(a)) from (values('aaaa'),('bbbb'),('cccc')) g(a);
+select concat_ws(',', collect_list(a)) from (values('aaaa'),(null),('bbbb'),('cccc')) g(a);
+select concat_ws('AB', collect_list(a)) from (values(null),(null),('bbbb'),('cccc')) g(a);
+-- The result is different: collect_list returns an empty list, but string_agg results NULL
+select concat_ws(',', collect_list(a)) from (values(null),(null)) g(a);
+
+-- check some implicit casting cases, as per bug #5564
+select concat_ws(',', sort_array(array_distinct(collect_list(f1)))) from varchar_tbl;
+-- select string_agg(distinct f1::text, ',' order by f1) from varchar_tbl;  -- not ok
+-- select string_agg(distinct f1, ',' order by f1::text) from varchar_tbl;  -- not ok
+-- select string_agg(distinct f1::text, ',' order by f1::text) from varchar_tbl;  -- ok
+
+-- Skip these tests because we do not have a bytea type
+-- string_agg bytea tests
+-- create table bytea_test_table(v bytea);
+
+-- select string_agg(v, '') from bytea_test_table;
+
+-- insert into bytea_test_table values(decode('ff','hex'));
+
+-- select string_agg(v, '') from bytea_test_table;
+
+-- insert into bytea_test_table values(decode('aa','hex'));
+
+-- select string_agg(v, '') from bytea_test_table;
+-- select string_agg(v, NULL) from bytea_test_table;
+-- select string_agg(v, decode('ee', 'hex')) from bytea_test_table;
+
+-- drop table bytea_test_table;
+
+-- [SPARK-27986] Support Aggregate Expressions
```

Comment:
I personally think this title name is a bit confusing cuz this is a general term for aggregates in Spark? https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala#L49

Reply:
Renamed it to `Support Aggregate Expressions with filter`.